homeassistant/components/sma/config_flow.py

```diff
@@ -0,0 +1,133 @@
+"""Config flow for the sma integration."""
+import logging
+
+import aiohttp
+import pysma
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_SENSORS,
+    CONF_SSL,
+    CONF_VERIFY_SSL,
+)
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+
+from .const import CONF_CUSTOM, CONF_GROUP, DEVICE_INFO, GROUPS
+from .const import DOMAIN  # pylint: disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def validate_input(hass: core.HomeAssistant, data: dict):
+    """Validate the user input allows us to connect."""
+    session = async_get_clientsession(hass, verify_ssl=data[CONF_VERIFY_SSL])
+
+    protocol = "https" if data[CONF_SSL] else "http"
+    url = f"{protocol}://{data[CONF_HOST]}"
+
+    sma = pysma.SMA(session, url, data[CONF_PASSWORD], group=data[CONF_GROUP])
+
+    if await sma.new_session() is False:
+        raise InvalidAuth
+
+    device_info = await sma.device_info()
+
+    if not device_info:
+        raise CannotRetrieveDeviceInfo
```

Comment:
Do we need to close the session in this case before raising?

Reply:
Correct. We should close the session before raising the exception (206044a)